core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -0,0 +1,537 @@
+                self.attempted_removals.add(removal_candidates)
+                return acceptance_candidates
+        return importance_df.index.tolist()
+
+    def compute_expected_fi_time_single(self, X_fi: pd.DataFrame, model_predict_time: float, n_subsample: int, n_total_sample: int) -> float:
+        """
+        Return the expected time to compute permutation feature importance for a single feature on n_total_sample datapoints.
+        """
+        n_subsample = min(n_subsample, len(X_fi))
+        n_shuffle = min(np.ceil(n_total_sample / n_subsample).astype(int), self.max_n_shuffle)
+        return self.safety_time_multiplier * model_predict_time * (n_subsample / len(X_fi)) * n_shuffle
+
+    def compute_time_budget_fi(self, X_fi: pd.DataFrame, n_subsample: int, **kwargs):
+        """
+        Return the time that a single feature importance computation round can take up to. Currently the minimum of 5 minutes
+        and the time it takes to fully evaluated minimum of 50 features or the number of features in the dataset.
+        """
+        min_fi_samples = kwargs.get('min_fi_samples', 10000)
+        max_fi_samples = kwargs.get('max_fi_samples', 100000)
+        n_total_samples = max(min_fi_samples, min(max_fi_samples, len(X_fi)))
+        fi_time_single = self.compute_expected_fi_time_single(X_fi, self.model_predict_time, n_subsample, n_total_samples)
+        return min(fi_time_single * min(len(X_fi.columns), 50), self.max_time_budget_fi)
+
+    def fit_score_model(self, X: pd.DataFrame, y: pd.Series, X_val: pd.DataFrame, y_val: pd.Series,
+                        features: List[str], model_name: str, **kwargs) -> Tuple[AbstractModel, float, float]:
+        """
+        Fits and scores a model over the given feature subset (ex. features remaining in a particular feature pruning round).
+        Updates self.time_limit. Returns the fitted model, its score, and time elapsed. If this is the first time we are fitting
+        a model in the pruning procedure, save time and score statistics.
+        """
+        time_start = time.time()
+        model = self.base_model.convert_to_template()
+        X = X[features]
+        X_val = None if self.is_bagged else X_val[features]
+        model.rename(model_name)
+        if 'time_limit' in kwargs:
+            kwargs['time_limit'] = self.time_limit
+        model.fit(X=X, y=y, X_val=X_val, y_val=y_val, **kwargs)
+        fit_time = time.time() - time_start
+        time_start = time.time()
+        score = model.score_with_oof(y) if self.is_bagged else model.score(X=X_val, y=y_val)
+        predict_time = time.time() - time_start
+        self.time_limit = self.time_limit - (fit_time + predict_time)
+        if self.model_fit_time is None:
+            self.model_fit_time = fit_time
+        if self.model_predict_time is None:
+            self.model_predict_time = predict_time
+        if self.original_val_score is None:
+            self.original_val_score = score
```

Comment:
Why? This is weird considering you originally require the model to be fit in init.